modules/tide_content_collection/src/Plugin/Field/FieldWidget/ContentCollectionConfigurationWidget.php

```diff
@@ -1279,50 +1304,99 @@ protected function buildFiltersTab(FieldItemListInterface $items, $delta, array
       if (!empty($json_object['interface']['filters']['fields'])) {
         $field_data = array_combine(array_column($json_object['interface']['filters']['fields'], 'elasticsearch-field'), $json_object['interface']['filters']['fields']);
       }
+      $sort_weight = -count($field_data);
+      $sorted_entity_reference_fields = [];
+      if (!empty($field_data)) {
+        foreach ($field_data as $field_id => $field_value) {
+          $field_id = (substr_compare($field_id, '_name', -strlen('_name')) === 0) ? str_replace('_name', '', $field_id) : $field_id;
+          if (!in_array($field_id, $entity_reference_fields)) {
+            $sorted_entity_reference_fields[$field_id] = $entity_reference_fields[$field_id];
+          }
+        }
+        if (!empty($sorted_entity_reference_fields)) {
+          $entity_reference_fields = array_merge($sorted_entity_reference_fields, $entity_reference_fields);
+        }
+      }
+      $element['tabs']['filters']['interface_filters']['advanced_filters']['items'] = [
+        '#type' => 'table',
+        '#tableselect' => FALSE,
+        '#attributes' => ['class' => ['advanced-filters-drag']],
+        '#tabledrag' => [
+          [
+            'action' => 'order',
+            'relationship' => 'sibling',
+            'group' => 'group-order-weight',
+          ],
+        ],
+      ];
       foreach ($entity_reference_fields as $field_id => $field_label) {
-        $element['tabs']['filters']['interface_filters']['advanced_filters'][$field_id]['allow'] = [
-          '#type' => 'checkbox',
-          '#title' => $field_label,
-          '#default_value' => isset($field_data[$field_id]) ? TRUE : FALSE,
-          '#weight' => $weight++,
-        ];
-        $element['tabs']['filters']['interface_filters']['advanced_filters'][$field_id][$field_id . '_label'] = [
-          '#title' => $this->t('Label'),
-          '#type' => 'textfield',
-          '#default_value' => $field_data[$field_id]['options']['label'] ?? '',
-          '#weight' => $weight++,
-          '#states' => [
-            'visible' => [
-              ':input[name="' . $this->getFormStatesElementName('tabs|filters|interface_filters|advanced_filters|' . $field_id . '|allow', $items, $delta, $element) . '"]' => ['checked' => TRUE],
+        if (!empty($validated_entity_reference_fields[$field_id])) {
+          $field_id = (substr_compare($field_id, '_name', -strlen('_name')) === 0) ? str_replace('_name', '', $field_id) : $field_id;
```

Comment:
@joshua-salsadigital if `$field_id = 'field_name_something_name'`, your code changes it to `field_something` which is not correct.

Reply:
Please use preg_match and preg_replace to ensure the correct logic
```
preg_match('/\_name$/', $field_id)
preg_replace('/\_name$/', '', $field_id)
```